service/src/main/java/bio/terra/externalcreds/services/PassportService.java

```diff
@@ -0,0 +1,73 @@
+package bio.terra.externalcreds.services;
+
+import bio.terra.common.db.ReadTransaction;
+import bio.terra.externalcreds.ExternalCredsException;
+import bio.terra.externalcreds.dataAccess.GA4GHPassportDAO;
+import bio.terra.externalcreds.models.GA4GHPassport;
+import bio.terra.externalcreds.models.LinkedAccount;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.oauth2.core.OAuth2RefreshToken;
+import org.springframework.security.oauth2.core.endpoint.OAuth2AccessTokenResponse;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class PassportService {
+
+  private final GA4GHPassportDAO passportDAO;
```

Comment:
`PassportService` does not use any transactional annotations so should not access DAOs directly. Pipe calls to `linkedAccountDAO` through `linkedAccountService` with proper transactional annotations in `LinkedAccountService`

I am also not convinced `PassportService` should exist.

Reply:
Yeah, we've updated the `PassportService` class a lot since Doug's original comment was made, and it now has transactional annotations. We decided to keep the class because it would be a more intuitive place to look for passport-related functionality than `LinkedAccountService`.